compiler/sem/transf.nim

```diff
@@ -550,22 +509,19 @@ proc transformConv(c: PTransf, n: PNode): PNode =
       result = transformSons(c, n)
     else:
       # generate a range check:
-      if dest.kind == tyInt64 or source.kind == tyInt64:
-        result = newTransNode(nkChckRange64, n, 3)
-      else:
-        result = newTransNode(nkChckRange, n, 3)
-      dest = skipTypes(n.typ, abstractVar)
-      result[0] = transform(c, n[1])
-      result[1] = newIntTypeNode(firstOrd(c.graph.config, dest), dest)
-      result[2] = newIntTypeNode(lastOrd(c.graph.config, dest), dest)
+      let rangeDest = skipTypes(n.typ, abstractVar)
+      result = newTreeIT(
+        if dest.kind == tyInt64 or source.kind == tyInt64: nkChckRange64 else: nkChckRange,
```

Comment:
```suggestion
        if tyInt64 in {dest.kind, source.kind}: nkChckRange64 else: nkChckRange,
```

A bit shorter, not necessarily clearer, might just be nicer if it was lifted into the let section above, then the rest reads like data declaration (no conditional logic).